Always allow rebalancing by default

DaveCTurner opened the pull request:
Today `cluster.routing.allocation.allow_rebalance` defaults to
`indices_all_active` which blocks all rebalancing moves while the
cluster is in `yellow` or `red` health. This was appropriate for the
legacy allocator which might do too many rebalancing moves otherwise.
The desired-balance allocator has better support for rebalancing a
cluster that is not in `green` health, and expects to be able to
rebalance some shards away from over-full nodes to avoid allocating
shards to undesirable locations in the first place. This commit changes
the default `allow_rebalance` setting to `always`.

DaveCTurner commented:
Opening as a draft for (a) CI time and (b) to sort out the changelogs.

elasticsearchmachine commented:
Hi @DaveCTurner, I've created a changelog YAML for you. *Note* that since this PR is labelled `release highlight`, you need to update the changelog YAML to fill out the extended information sections.

DaveCTurner committed:
Always allow rebalancing by default

Today `cluster.routing.allocation.allow_rebalance` defaults to
`indices_all_active` which blocks all rebalancing moves while the
cluster is in `yellow` or `red` health. This was appropriate for the
legacy allocator which might do too many rebalancing moves otherwise.
The desired-balance allocator has better support for rebalancing a
cluster that is not in `green` health, and expects to be able to
rebalance some shards away from over-full nodes to avoid allocating
shards to undesirable locations in the first place. This commit changes
the default `allow_rebalance` setting to `always`.

elasticsearchmachine commented:
Pinging @elastic/es-distributed (Team:Distributed)
